### PR TITLE
provider/google: Add support for session_affinity to google_compute_region_backend_service

### LINKED ISSUE #11199: Feature Request: Add `session_affinity` property on `google_compute_region_backend_service`
Hey, this is a feature request. Details below

### Terraform Version
Terraform v0.8.4

### Affected Resource(s)
* google_compute_region_backend_service

### Terraform Configuration Files Example
```hcl
resource "google_compute_region_backend_service" "bes" {
  name             = "bes-prgall"
  description      = "This resource is managed by Terraform, do not edit manually!"
  protocol         = "TCP"
  timeout_sec      = "30"
  health_checks    = ["${google_compute_health_check.hck.self_link}"]
  session_affinity = "CLIENT_IP"
  region           = "${var.region}"

  backend {
    group = "${google_compute_instance_group_manager.igm.instance_group}"
  }

}
```

### Expected Behavior
I would like to use the `session_affinity` attribute on a `google_compute_region_backend_service`.
I know that this is a new feature on google's part, so I'm just adding some backlog here. I'll see if I could contribute, although I'm not used of coding in go ^^'

### Actual Behavior
```
Errors:

  * google_compute_region_backend_service.bes-prgall: : invalid or unknown key: session_affinity

```

### References
I have found none, I'm sorry if there is an existing issue

### PR DESCRIPTION
**Pull request by cblecker**
```
$ make testacc TEST=./builtin/providers/google TESTARGS='-run=TestAccComputeRegionBackendService_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/16 13:44:45 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccComputeRegionBackendService_ -timeout 120m
=== RUN   TestAccComputeRegionBackendService_basic
--- PASS: TestAccComputeRegionBackendService_basic (67.05s)
=== RUN   TestAccComputeRegionBackendService_withBackend
--- PASS: TestAccComputeRegionBackendService_withBackend (105.11s)
=== RUN   TestAccComputeRegionBackendService_withBackendAndUpdate
--- PASS: TestAccComputeRegionBackendService_withBackendAndUpdate (131.22s)
=== RUN   TestAccComputeRegionBackendService_withSessionAffinity
--- PASS: TestAccComputeRegionBackendService_withSessionAffinity (66.38s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/google	369.775s
```

Fixes #11199.

### CODE REVIEW

**Comment by cblecker** [comment]
cc: @evandbrown @danawillow FYI :)

File: builtin/providers/google/resource_compute_region_backend_service_test.go [hunk 1]
```diff
@@ -114,6 +114,36 @@ func TestAccComputeRegionBackendService_withBackendAndUpdate(t *testing.T) {
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeRegionBackendServiceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeRegionBackendService_withSessionAffinity(serviceName, checkName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeRegionBackendServiceExists(
+						"google_compute_region_backend_service.foobar", &svc),
```
Comment (danawillow) [inline builtin/providers/google/resource_compute_region_backend_service_test.go]: Can you also check here that the new region backend service not just exists but has the correct session affinity?

Comment (cblecker) [inline builtin/providers/google/resource_compute_region_backend_service_test.go]: Fixed.

**Comment by cblecker** [comment]
Result of modified acceptance test:
```
$ make testacc TEST=./builtin/providers/google TESTARGS='-run=TestAccComputeRegionBackendService_withSessionAffinity'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/17 16:28:16 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccComputeRegionBackendService_withSessionAffinity -timeout 120m
=== RUN   TestAccComputeRegionBackendService_withSessionAffinity
--- PASS: TestAccComputeRegionBackendService_withSessionAffinity (44.65s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/google	44.668s
```

**Comment by zopanix** [comment]
Is it possible to have an ETA when this will be merged ? And is it possible to know when next release will be ? Just to know if I need to build terraform locally to have this feature :)

**Comment by danawillow** [comment]
LGTM, thanks @cblecker! Merging now. @zopanix I'm not sure when the next release will be.

**Comment by zopanix** [comment]
@danawillow Ok, thx for merging, I'll build terraofrm using master for the moment and will be waiting for next release :) I must say things are moving quickly here. That's always nice to have :)

**Comment by cblecker** [comment]
Thanks for the review and merge, @danawillow! Appreciated :)

@zopanix -- Minor revisions of Terraform are cut on average every two weeks. Last release was cut a week ago, so maybe another week or so? There are never guarantees, but it won't be long that's for sure.

**Pull request merged**